mlflow/catboost.py

```diff
@@ -168,14 +184,8 @@ def save_model(
     pyfunc.add_to_model(
         mlflow_model, loader_module="mlflow.catboost", env=_CONDA_ENV_FILE_NAME, **model_bin_kwargs,
     )
-
-    flavor_conf = {
-        _MODEL_TYPE_KEY: cb_model.__class__.__name__,
-        _SAVE_FORMAT_KEY: kwargs.get("format", "cbm"),
-        **model_bin_kwargs,
-    }
     mlflow_model.add_flavor(
-        FLAVOR_NAME, catboost_version=cb.__version__, **flavor_conf,
+        FLAVOR_NAME, catboost_version=cb.__version__, **flavor_conf, **model_bin_kwargs,
     )
```

Comment:
In accordance with https://github.com/mlflow/mlflow/pull/4518/files#r683751938, can we move this up to the part of the code right before we perform dependency resolution?